src/kudu/consensus/consensus_queue.cc

```diff
@@ -1620,6 +1626,19 @@ bool PeerMessageQueue::ResponseFromPeer(const std::string& peer_uuid,
                             ALL_REPLICAS,
                             peer);
 
+      // Update 'region_durable_index' if the response is from a peer outside
+      // our region
+      if (peer->last_received.index() > queue_state_.region_durable_index &&
+          peer->peer_pb.attrs().has_region() &&
```

Comment:
can we check whether we are always expecting region tag for all peers. If not we should update topology config checks on raft plugin side

Reply:
should this be abstracted out into a function like AdvanceQueueWatermark("region_durable")?